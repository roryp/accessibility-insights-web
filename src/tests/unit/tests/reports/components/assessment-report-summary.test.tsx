// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import {
    AssessmentReportSummary,
    AssessmentReportSummaryProps,
} from 'reports/components/assessment-report-summary';

describe('AssessmentReportSummary', () => {
    describe('render', () => {
        test('Correct composition', () => {
            const reportData: AssessmentReportSummaryProps = {
                summary: {
                    reportSummaryDetailsData: [],
                },
            } as AssessmentReportSummaryProps;

            const testObject = new AssessmentReportSummary(reportData);
            const actual = testObject.render();

            expect(actual).toMatchSnapshot();
        });
    });
});
