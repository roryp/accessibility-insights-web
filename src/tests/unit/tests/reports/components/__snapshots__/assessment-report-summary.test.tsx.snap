// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssessmentReportSummary render Correct composition 1`] = `
<div
  className="assessment-report-summary"
>
  <h2>
    Summary
  </h2>
  <OutcomeSummaryBar
    allOutcomeTypes={
      Array [
        "pass",
        "incomplete",
        "fail",
      ]
    }
    countSuffix="%"
  />
  <h3
    className="test-details-text"
  >
    Test details
  </h3>
  <AssessmentSummaryDetails
    testSummaries={Array []}
  />
</div>
`;
